subprojects/core-api/src/main/java/org/gradle/api/plugins/internal/DelegatingConventionInterface.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.gradle.api.plugins.internal;
+
+import org.gradle.api.Action;
+import org.gradle.api.UnknownDomainObjectException;
+import org.gradle.api.plugins.ExtensionContainer;
+import org.gradle.api.plugins.ExtensionsSchema;
+import org.gradle.api.plugins.ExtraPropertiesExtension;
+import org.gradle.api.reflect.TypeOf;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * Implementation of the custom convention interface that delegates all ExtensionContainer methods
+ * to another ExtensionContainer implementation.
```

Comment:
I think we need to document why we have this interface. We should explain that we have to support plugins compiled against old Gradle version that still use conventions.